core-bundle/src/Image/Studio/MetaData.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Studio;
+
+use Contao\StringUtil;
+
+/**
+ * @psalm-immutable
+ */
+final class MetaData
+{
+    public const PROPERTY_DOMINANT_VALUES = 'overwriteMeta';
+    public const PROPERTY_FULLSIZE = 'fullsize';
+    public const PROPERTY_MARGIN = 'margin';
+    public const PROPERTY_FLOATING = 'floating';
+
+    public const VALUE_ALT = 'alt';
+    public const VALUE_CAPTION = 'caption';
+    public const VALUE_LINK_TITLE = 'linkTitle';
+    public const VALUE_TITLE = 'title';
+    public const VALUE_URL = 'link';
+
+    /**
+     * @var array
+     */
+    private $values;
+
+    /**
+     * @var array
+     */
+    private $properties;
+
+    public function __construct(array $values, array $properties = [])
+    {
+        $this->values = $values;
+        $this->properties = array_unique($properties);
+    }
+
+    public function withOther(self $metaData, bool $forceMergingValues = false): self
+    {
+        $values = $forceMergingValues || false !== $metaData->hasDominantValues() ?
```

Comment:
I know it's still WIP but everything could do with a bit more comments, maybe? 😄 
The "dominant values" concept looks like a code smell to me but maybe I'm wrong. But without comments it's kinda hard to understand what your brain was thinking^^

Reply:
True. I'm not yet 100% convinced about this part as well. :smile: But I'll sure add in some comments with the next iteration.

The Idea behind this was roughly to allow merging two `MetaData` items and let the items themselves decide how to handle this (mapping the 'overwrite metadata' functionality) - kind of like a priority. That's also why it's split into properties (meta meta data) and values.